ipatests/test_integration/test_acme.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+import time
+
+from ipalib.constants import IPA_CA_RECORD
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+
+class TestACME(IntegrationTest):
+    """
+    Test the FreeIPA ACME service by using ACME clients on a FreeIPA client.
+
+    Right now the only thing we test is the Certbot client using
+    http-01 challenge with Certbot's standalone HTTP server.
+    We can add tests for DNS challenges later.
+
+    """
+
+    num_replicas = 0
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install certbot before ipa-client-install in case of IPA DNS problems
+        cls.clients[0].run_command(['dnf', 'install', '-y', 'certbot'])
```

Comment:
Hello @frasertweedale 
Could you change this to install with `pip` so that it doesn't break when run on rhel?

Reply:
@mrizwan93 is it a requirement that these integration tests work both upstream and downstream?  I would rather stick to the Fedora package for upstream Ci.

Reply:
Yes, we run integration tests downstream.

Reply:
@mrizwan93 this turns out to be a bit of a problem due to incompatibilities between recent versions of *certbot* available from PyPI and packages in RHEL (in particular *python-cryptography*, which is currently at 2.3 in RHEL 8).

Can we use EPEL repository instead (certbot is in EPEL).  Or if not, can we just install specific package(s) from EPEL?  (IMO the latter case is similar to `pip install` except that packagers already solved the hard problems!)

Reply:
@mrizwan93 alternatively we can disable this test for RHEL and implement another integration test using *mod_md*, which is supported on RHEL.  We can also implement simple queries using *curl* to ensure ACME service is responding or not responding according to whether it is enabled/disabled.  We want to implement such tests anyway - so if we do that, maybe it is not a big deal to skip the *certbot* test on RHEL.  Let me know what you think.

Reply:
@frasertweedale EPEL sounds good :)
ACK.

Reply:
@mrizwan93 well what do you know, the EPEL certbot package is also broken against *python-cryptography-2.3*.  I think that it is better to disable tests that require *certbot* on RHEL, and proceed with implementing tests that use *mod_md* which is available on RHEL.

Reply:
@mrizwan93 I implemented mod_md client test which will execute on RHEL.  From point of view of IPA ACME service, it sees an arbitrary ACME client responding to the `http-01` challenge, which is no different from what it would see for the *certbot* standalone scenario.  So I think it is enough.

There are still DNS tests we could implement, but those require certbot so we're kind of stuck doing that on RHEL (unless we find another ACME client that can do the DNS challenges, that we can use on RHEL).

Reply:
Relying on EPEL for such critical parts of our test suite might make it more brittle ; packages can get removed from EPEL {dist-git branches, repos} at RHEL minor release time if they are newly shipped in RHEL. Possible breakage:
- CI using RHEL n.m-1 images suddenly flips red because the required package is not available in any repo.
- There is no guarantee RHEL will ship the version that was in EPEL.
In general things work, but sometimes they don't, and the timing can be inconvenient.
So +1 to switching to mod_md from me.